### PR TITLE
add test for web-accessibility

### LINKED ISSUE #248: Add web accessiblity in FAQ section
As currently spacebar is not accessible.
After pressing the tab How do I contribute to RDS? the browser didn't show  their child element after pressing enter and the space bar it can show it child element.
Screenshot (152)

### PR DESCRIPTION
**Pull request by vinit717**
close #248

### CODE REVIEW

File: Sample-test/sample.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
+// const addTodo = () => {
+//   const newTodoInput = document.getElementById('newTodoInput');
+//   let currentTodoList = document.getElementById('todoList').innerHTML;
+//   currentTodoList += `<li>${newTodoInput.value}</li>`;
+//   document.getElementById('todoList').innerHTML = currentTodoList;
+//   newTodoInput.value = '';
+// };
+// document.getElementById('addTodoBtn').addEventListener('click', addTodo);
+
+// export default addTodo;
```
Comment (heyrandhir) [inline Sample-test/sample.js:10]: If these are not needed then we can remove this file

File: Sample-test/sample.test.js [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
+//       <ol id="todoList"></ol>
+//     `;
+
+//   const newTodoInput = document.getElementById('newTodoInput');
+//   const addTodoBtn = document.getElementById('addTodoBtn');
+//   const todolist = document.getElementById('todoList');
+//   newTodoInput.value = 'welcome to RDS';
+//   addTodoBtn.click();
+//   expect(todolist.innerHTML).toBe('<li>welcome to RDS</li>');
+// });
```
Comment (heyrandhir) [inline Sample-test/sample.test.js:15]: If these are not needed then we can remove this file

File: Sample-test/sample.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+<!-- <!DOCTYPE html>
 <html lang="en">
   <head>
     <meta charset="UTF-8" />
```
Comment (heyrandhir) [inline Sample-test/sample.html:4]: If this is commented this can also be removed

File: js/faq.test.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    ancTag.setAttribute('href', '/example');
+    ancTag.click = jest.fn();
+    document.body.appendChild(ancTag);
+    event = { key: ' ', preventDefault: jest.fn() };
+    // event = new KeyboardEvent('keydown', { key: ' ' });
+  });
+
+  // afterEach(() => {
+  //        spy.mockRestore();
+  //     });
```
Comment (heyrandhir) [inline js/faq.test.js:18]: I think we should mockRestore after each test

File: js/faq.test.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    window.addEventListener('keydown', event);
+    expect(event.preventDefault).not.toHaveBeenCalled();
+  });
+
+  it('should not call ancTag.click method when key is not space', () => {
+    event.key = 'Enter';
+    window.addEventListener('keydown', event);
+    expect(ancTag.click).not.toHaveBeenCalled();
+  });
+});
```
Comment (heyrandhir) [inline js/faq.test.js:42]: NIT : since most of the code is same for both they can be clubbed.

File: js/faq.test.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        ],
+      },
+    },
+  ];
+  const target = faqLinks[0];
+
+  removeFocusForOthers(target);
+
+  faqLinks.forEach((faqLink, index) => {
+    if (index === 1) {
```
Comment (heyrandhir) [inline js/faq.test.js:61]: Since there is only 1 element in `faqLinks` index will never be 1

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: Sample-test/sample.js (hunk 1 above)
Comment (vinit717) [reply Sample-test/sample.js:10]: It basically for pr of test setup to show how we suppose to write a test

File: Sample-test/sample.html (hunk 3 above)
Comment (vinit717) [reply Sample-test/sample.html:4]: yes

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: js/faq.test.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  afterEach(() => {
+    spy.mockRestore();
+  });
+
+  it('should not call preventDefault and ancTag.click method  when key is not space', () => {
+    event.key = 'Enter';
+    window.addEventListener('keydown', event);
+    expect(event.preventDefault).not.toHaveBeenCalled();
+  });
+});
```
[the remark this answers is not in the document]
Comment (vinit717) [reply js/faq.test.js:24]: I'm finding about  it and I will have to change the whole code to do that. Later on I will raise another PR to do this.

**Review by heyrandhir — APPROVED** [review]

File: faq.html [hunk 8]
```diff
@@ -38,10 +38,11 @@
       </div>
       <div class="nav-login">
         <a
+          tabindex="1"
           class="btn-login"
           href="https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97"
         >
-          <button class="btn-login-text">
+          <button tabindex="1" class="btn-login-text">
```
Comment (DeRaowl) [inline faq.html:45]: button itself is a semantic element and we should not change the order of tabIndex. Can you provide the reason for changing tabindex order?

File: faq.html [hunk 9]
```diff
@@ -53,14 +54,20 @@
...
+            <img class="user-profile-pic" />
+          </a>
         </div>
       </div>
 
       <ul class="nav-menu">
         <li class="nav-logo-li">
-          <a href="https://www.realdevsquad.com/" class="logo-link">
+          <a
+            tabindex="1"
```
Comment (DeRaowl) [inline faq.html:67]: why not tabIndex=0?

File: faq.html [hunk 10]
```diff
@@ -69,22 +76,39 @@
             />
           </a>
         </li>
-        <li class="home-tab"><a href="http://realdevsquad.com/">Home</a></li>
+        <li class="home-tab">
+          <a tabindex="1" href="http://realdevsquad.com/">Home</a>
```
Comment (DeRaowl) [inline faq.html:80]: Same here. Avoid using positive tab index

File: faq.html [hunk 11]
```diff
@@ -69,22 +76,39 @@
...
+        </li>
         <li>
-          <a class="active-tab" href="https://welcome.realdevsquad.com/">
+          <a
+            class="active-tab"
+            tabindex="1"
```
Comment (DeRaowl) [inline faq.html:85]: https://github.com/reactjs/react-a11y/blob/master/docs/rules/tabindex-no-positive.md

Read up more here.

File: faq.html [hunk 12]
```diff
@@ -69,22 +76,39 @@
...
+          >
             Welcome
           </a>
         </li>
-        <li><a href="https://www.realdevsquad.com/events.html">Events</a></li>
-        <li><a href="https://members.realdevsquad.com/">Members</a></li>
-        <li><a href="https://crypto.realdevsquad.com/">Crypto</a></li>
-        <li><a href="https://status.realdevsquad.com/">Status</a></li>
+        <li>
+          <a tabindex="1" href="https://www.realdevsquad.com/events.html"
```
Comment (DeRaowl) [inline faq.html:92]: Same here.

File: faq.html [hunk 13]
```diff
@@ -69,22 +76,39 @@
...
+            >Events</a
+          >
+        </li>
+        <li>
+          <a tabindex="1" href="https://members.realdevsquad.com/">Members</a>
```
Comment (DeRaowl) [inline faq.html:97]: Observed that most of the places we have used tabIndex="1". Can we replace it with "0" or can you provide the reason for using "1"

File: faq.html [hunk 14]
```diff
@@ -127,11 +151,11 @@ <h3 class="question">What is Real Dev Squad?</h3>
               </p>
             </div>
           </div>
-          <div class="faq">
-            <a class="faq_link" href="#How-do-I-contribute-to-RDS">
-              <div class="faq__title">
+          <div class="faq" type="button">
```
Comment (DeRaowl) [inline faq.html:154]: If you are using type="button" for div then we might need to give aria-label, keydown event, tabIndex. Can we make use of button instead?

File: faq.html [hunk 15]
```diff
@@ -127,11 +151,11 @@ <h3 class="question">What is Real Dev Squad?</h3>
...
+            <a tabindex="1" class="faq_link" href="#How-do-I-contribute-to-RDS">
+              <div class="faq__title" type="button" role="button">
```
Comment (DeRaowl) [inline faq.html:156]: Same here

File: faq.html [hunk 16]
```diff
@@ -142,39 +166,40 @@ <h3 class="question">How do I contribute to RDS?</h3>
 
               <!-- prettier-ignore -->
               <ul class="faq__list">
-              <li><a href="https://github.com/Real-Dev-Squad/website-welcome" target="_blank"
+              <li><a tabindex="-1" href="https://github.com/Real-Dev-Squad/website-welcome" target="_blank"
```
Comment (DeRaowl) [inline faq.html:169]: Why are we using tab-index -1?

File: js/faq.js [hunk 17]
```diff
@@ -1,8 +1,37 @@
 const faqLinks = document.querySelectorAll('.faq_link');
 const currentLocation = window.location.hash;
 
+// to open accordian by pressing spacebar
+window.addEventListener('keydown', (e) => {
```
Comment (DeRaowl) [inline js/faq.js:5]: Why not document?

File: js/faq.js [hunk 18]
```diff
@@ -1,8 +1,37 @@
...
+  const ancTag = document.querySelector(
+    `a[href="${e?.target?.attributes?.href?.value}"]`,
+  );
+  if (e.key === ' ' && ancTag) {
```
Comment (DeRaowl) [inline js/faq.js:9]: '    ' signify the case here?

File: js/faq.js [hunk 19]
```diff
@@ -1,8 +1,37 @@
...
+    if (faqLink.getAttribute('href') != target.getAttribute('href')) {
+      const faqTextSiblingElement = faqLink.nextElementSibling;
+      const ancTag = faqTextSiblingElement.querySelectorAll(`a`);
+      ancTag.forEach((element) => {
+        element.setAttribute('tabindex', '-1');
+      });
+    }
+  });
+};
+export default removeFocusForOthers;
```
Comment (DeRaowl) [inline js/faq.js:27]: export should be at last

**Review by DeRaowl — CHANGES_REQUESTED** [review]
As it is draft PR, can we make sure we are getting this fixed in final one

File: js/faq.js (hunk 19 above)
Comment (vinit717) [reply js/faq.js]: done

File: js/faq.js (hunk 18 above)
Comment (vinit717) [reply js/faq.js:9]: for open  accordion with space bar

File: faq.html (hunk 8 above)
Comment (vinit717) [reply faq.html]: removed tabindex from button

File: faq.html (hunk 9 above)
Comment (vinit717) [reply faq.html:68]: To open the accordion as our need

File: faq.html [hunk 20]
```diff
@@ -6,6 +6,7 @@
 
     <title>FAQ</title>
 
+    <link rel="stylesheet" type="text/css" href="./css/navbar.css" />
```
Comment (DeRaowl) [inline faq.html:9]: Why we have same link twice?

File: js/faq.js (hunk 18 above)
Comment (DeRaowl) [reply js/faq.js:9]: use e.code == "Space" ||      
      e.keyCode == 32 for spacebar key event

**Review by DeRaowl — CHANGES_REQUESTED** [review]

File: faq.html (hunk 20 above)
Comment (vinit717) [reply faq.html]: resolved

File: faq.html (hunk 13 above)
Comment (vinit717) [reply faq.html:97]: it basically depend upon the scenerio what we want

File: faq.html (hunk 14 above)
Comment (vinit717) [reply faq.html:154]: The whole earlier would be refactor in another Pr

File: js/faq.js (hunk 17 above)
Comment (vinit717) [reply js/faq.js:5]: Space bar event is not work on document

File: faq.html (hunk 16 above)
Comment (vinit717) [reply faq.html:169]: To hide the the links when accordion is closed

**Review by DeRaowl — APPROVED** [review]
Approving this but make sure we fix the accessible issues in the upcoming PR.

**Pull request merged**